src/player_activity.cpp

```diff
@@ -318,6 +318,11 @@ void player_activity::finish( player *p )
             p->do_read(&(p->i_at(position)));
             if (type == ACT_NULL) {
                 add_msg(_("You finish reading."));
+
+                // below code added in to support reading from ground/vehicles
+                if ( values[1] != FRM_INV ) {
+                    g->drop(p->inv.size()-1);
```

Comment:
Hi BrianLefler,

You are right. It could be a bug as there is no guarantee that the last inventory item is the book.

Can I pass the book item address (last inventory item in game.read()) into player_activity.str_values?
Then pass it into inventory.position_by_item() and thus dropping the correct item